actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -438,6 +438,10 @@ def ssl?
       super || scheme == "wss"
     end
 
+    def trustworthy?
+      ssl? || host.end_with?(".onion")
```

Comment:
instead of doing this as a global change for all Rails apps, would it make sense to expose this as a config so that relevant apps can define one or more trusted domains?

But then, `.onion` is a special use domain, so doing this as a global change might be fine too.